### ISSUE TITLE
Scans broken in Buildkite

### ISSUE THREAD

**Issue by mogopz** [semgrep/semgrep#5168]
**Describe the bug**
A change was introduced in 0.89.0 which seems to have broken our builds in Buildkite. We run Buildkite's [elastic-ci-stack](https://github.com/buildkite/elastic-ci-stack-for-aws) which is their recommended way of managing agents in AWS. When the Buildkite agent running on an instance clones a repository the files are all owned by the `buildkite-agent` user and since that is different to the root user that your Docker image runs as we get this error:

```
=== Running: semgrep ci
--
  | Scan environment:
  | versions    - semgrep 0.91.0 on python 3.10.4
  | environment - running in environment git, triggering event is unknown
  | semgrep.dev - authenticated as localz
  |  
  | Fetching configuration from semgrep.dev
  | Traceback (most recent call last):
  | File "/usr/local/lib/python3.10/site-packages/semgrep/commands/ci.py", line 298, in ci
  | scan_handler.start_scan(metadata.to_dict())
  | File "/usr/local/lib/python3.10/site-packages/semgrep/meta.py", line 128, in to_dict
  | commit_title = subprocess.check_output(
  | File "/usr/local/lib/python3.10/subprocess.py", line 420, in check_output
  | return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  | File "/usr/local/lib/python3.10/subprocess.py", line 524, in run
  | raise CalledProcessError(retcode, process.args,
  | subprocess.CalledProcessError: Command '['git', 'show', '-s', '--format=%B']' returned non-zero exit status 128.
  | Could not start scan Command '['git', 'show', '-s', '--format=%B']' returned non-zero exit status 128.
```

You have a PR ([#5044](https://github.com/returntocorp/semgrep/pull/5044)) which introduces a `GITHUB_WORKSPACE` env var as a workaround and I can confirm this DOES work but I'm not sure it's the best solution or naming since `GITHUB_WORKSPACE` is a default env var for Github actions but issue will apply to many people who are using different CI systems.

At the very least the [example docs](https://semgrep.dev/docs/semgrep-ci/sample-ci-configs/#buildkite) should be updated to reflect (I'm more than happy to open a PR there!). This is what I've tested and it seems to work:

```yaml
steps:
  - label: ":semgrep: Running static code analysis"
    command: semgrep ci
    plugins:
      - docker#v3.13.0:
          image: returntocorp/semgrep:0.91.0
          workdir: /<org_name>/<repo_name>
          environment:
            - "SEMGREP_APP_TOKEN=${SEMGREP_APP_TOKEN}"
            - "GITHUB_WORKSPACE=/<org_name>/<repo_name>"
            - "SEMGREP_JOB_URL=${BUILDKITE_BUILD_URL}"
            - "SEMGREP_BRANCH=${BUILDKITE_BRANCH}"
            - "SEMGREP_REPO_NAME=/<org_name>/<repo_name>"
            - "SEMGREP_REPO_URL=https://github.com/<org_name>/<repo_name>"
```

**What is the priority of the bug to you?**

- [ ] P0: blocking your adoption of Semgrep or workflow
- [ ] P1: important to fix or quite annoying
- [x] P2: regular bug that should get fixed

**Comment by r2c-demo** [comment]
This issue is synced in Linear at https://linear.app/r2c/issue/PA-1263/scans-broken-in-buildkite. _Note_: this link is for r2c use only and is not accessible publicly.

**Comment by brendongo** [comment]
Thanks for the write up @mogggggg 

We tested on GithubActions and GitlabCI and did not see the same error in Gitlab so incorrectly assumed this was specific to github :D . We could just always run this workaround when we are inside a CI context maybe (I think all providers set CI to true in that case).

**Comment by underyx** [comment]
@brendongo I think we can also unconditionally run the command whenever we're in a docker container (either semgrep-agent or semgrep). The CVE allows an attack if somebody else controls an ancestor of your current working directory. But with our docker images, you mount the source code right into the working directory you use (since you decided what to mount, that directory is trusted), and any parent directory would be supplied by r2c (who is also trusted).

The only way I think this could be exploited is if there's a user who mounts their whole file system into the docker container and then sets working directory to `--workdir=/mounted-fs/home/my-username/projects/some-app` or something. I've never seen someone do that before.

tl;dr: let's use the `SEMGREP_IN_DOCKER=1` env var?

**Comment by brendongo** [comment]
That makes sense

**Comment by mogopz** [comment]
> tl;dr: let's use the `SEMGREP_IN_DOCKER=1` env var?

Sounds good to me! Thanks for the quick replies 🙏

**Comment by brendongo** [comment]
This was fixed by https://github.com/returntocorp/semgrep/pull/5174 and should be in 0.92.0 @mogggggg 

We set that envvar automatically in our docker image so on 0.92.0 you shouldnt need to set  "GITHUB_WORKSPACE=/<org_name>/<repo_name>"

Let us know if this is still an issue

**Comment by mogopz** [comment]
> This was fixed by #5174 and should be in 0.92.0 @mogggggg

Awesome, thankyou! I've just tested and confirmed it works now
